packages/hooks/src/useLongPress/index.ts

```diff
@@ -0,0 +1,48 @@
+import { useRef } from 'react';
+import useBoolean from '../useBoolean';
+import useEventListener from '../useEventListener';
+import useLatest from '../useLatest';
+import type { BasicTarget } from '../utils/dom2';
+
+type EventType = MouseEvent | TouchEvent;
+
+function useLongPress(onLongPress: (event: EventType) => void, target: BasicTarget, delay: number) {
+  const onLongPressRef = useLatest(onLongPress);
+  const timer = useRef<ReturnType<typeof setTimeout>>();
+  const [state, { setTrue, setFalse }] = useBoolean(false);
+
+  let startTime = 0;
+  let endTime = 0;
+
+  useEventListener(
+    'mousedown',
+    (e) => {
+      e.preventDefault();
+
+      startTime = Date.now();
+      timer.current = setTimeout(() => onLongPressRef.current(e), delay);
+
+      setTrue();
+    },
+    { target },
+  );
+
+  useEventListener(
+    'mouseup',
+    (e) => {
+      e.preventDefault();
+
+      endTime = Date.now();
+      if (endTime - startTime < delay) {
+        clearTimeout(timer.current!);
+      }
```

Comment:
这里好像不用判断的，直接 clear 就行了

```suggestion
        clearTimeout(timer.current!);
```

Reply:
有道理，我去掉